cli/pa_analyze_command.go

```diff
@@ -413,6 +425,118 @@ func (cmd *PaAnalyzeCmd) checkLaggingStreamReplicas(r *archive.Reader) (checkSta
 	return Pass, nil
 }
 
+const cpuUsageThreshold = 0.9 // Warn if any server is using more than 90% of the available CPU
+// checkCpuUsage checks the CPU usage of all servers and alerts if any server is using more than 90% of the available CPU
+func (cmd *PaAnalyzeCmd) checkCpuUsage(r *archive.Reader) (checkStatus, error) {
+	serverTags := r.ListServerTags()
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	for _, serverTag := range serverTags {
+		serverName := serverTag.Value
+		var serverVarz server.Varz
+
+		if err := r.Load(&serverVarz, &serverTag, archive.TagServerVars()); errors.Is(err, archive.ErrNoMatches) {
+			cmd.logWarning("Artifact 'VARZ' is missing for server %s", serverName)
+			continue
+		} else if err != nil {
+			return Skipped, fmt.Errorf("failed to load VARZ for server %s: %w", serverName, err)
+		}
+
+		cpuUsageForASingleCore := serverVarz.CPU / float64(serverVarz.Cores)
+
+		if cpuUsageForASingleCore > cpuUsageThreshold {
+			examples.Addf("%s: %.0f%%", serverName, serverVarz.CPU)
+		}
+	}
+
+	if examples.Count() > 0 {
+		cmd.logIssue("Found servers with high CPU usage")
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+
+	return Pass, nil
+}
+
+const highCardinalityStreamsThreshold = 1_000_000 // Warn if any stream has more than 1,000,000 unique subjects
+// checkHighCardinalityStreams checks the number of unique subjects in streams and alerts if any stream has a high number of unique subjects
+func (cmd *PaAnalyzeCmd) checkHighCardinalityStreams(r *archive.Reader) (checkStatus, error) {
+	typeTag := archive.TagStreamDetails()
+	accountNames := r.GetAccountNames()
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	for _, accountName := range accountNames {
+		accountTag := archive.TagAccount(accountName)
+		streamNames := r.GetAccountStreamNames(accountName)
+
+		for _, streamName := range streamNames {
+			streamTag := archive.TagStream(streamName)
+
+			serverNames := r.GetStreamServerNames(accountName, streamName)
+			for _, serverName := range serverNames {
+				serverTag := archive.TagServer(serverName)
+
+				var streamDetails server.StreamDetail
+
+				if err := r.Load(&streamDetails, serverTag, accountTag, streamTag, typeTag); errors.Is(err, archive.ErrNoMatches) {
+					cmd.logWarning("Artifact 'STREAM_DETAILS' is missing for stream %s in account %s", streamName, accountName)
+					continue
+				} else if err != nil {
+					return Skipped, fmt.Errorf("failed to load STREAM_DETAILS for stream %s in account %s: %w", streamName, accountName, err)
+				}
+
+				if streamDetails.State.NumSubjects > highCardinalityStreamsThreshold {
+					examples.Addf("%s/%s: %d subjects", accountName, streamName, streamDetails.State.NumSubjects)
```

Comment:
This potentially adds N identical examples for the same stream, where N is the number of replicas.

Maybe just add one example per stream?
(after you find one problem, `continue` to the next stream, don't bother checking the other replicas)